app/controllers/navigation/Navigator.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.navigation
+
+import config.AppConfig
+import javax.inject.Inject
+import models.requests.JourneyRequest
+import models.responses.FlashKeys
+import play.api.mvc.{Result, Results}
+
+class Navigator @Inject()(appConfig: AppConfig) {
+
+  def goToDraftConfirmation()(implicit req: JourneyRequest[_]): Result = {
+    val updatedDateTime = req.cacheModel.updatedDateTime
+    val expiry = updatedDateTime.plusSeconds(appConfig.draftTimeToLive.toSeconds)
+    Results.Redirect(
+      controllers.declaration.routes.ConfirmationController.displayDraftConfirmation()
+    ).flashing(FlashKeys.expiryDate -> expiry.toEpochMilli.toString)
```

Comment:
Is this the best way of getting config data into the view?   Why not just give the view the AppConfig?

Reply:
So I think the actual declaration is meant to be out of memory at the point of the confirmation controller, but it needs to know the expiry date. 
So options are: 
- use session storage
- use flash storage (one request session storage)
- don't remove declaration from memory

Reply:
Didn't realise you'd lost the declaration at that point.

Reply:
Well, I'd missed the code to pull it from the session but I've added it now :D